gradle-plugin/src/main/kotlin/knote/plugin/KNotePlugin.kt

```diff
@@ -70,27 +72,95 @@ open class KNotePlugin : Plugin<Project> {
             }
         }
 
+        val notebookDir = project.rootDir.resolve("notebooks").apply { mkdirs() }
+
+        project.extensions.configure<IdeaModel> {
+            module {
+//                val debugLog = project.rootDir.resolve("debug.log")
+//                debugLog.writeText("adding '$notebookDir' to sourceDirs")
+
+                project.logger.lifecycle("adding '$notebookDir' to sourceDirs")
+                sourceDirs.add(notebookDir)
+                notebookDir
+                    .listFiles { _, name -> name.endsWith(".notebook.kts") }
+                    .forEach { scriptFile ->
+                        val id = scriptFile.name.substringBeforeLast(".notebook.kts")
+
+                        val generatedSrc = project.rootDir.resolve("build").resolve(".knote").resolve(id)
+                        val pagesSrc = project.rootDir.resolve("${id}_pages")
+
+                        sourceDirs.add(pagesSrc)
```

Comment:
are these for the maven repository?

Reply:
this was my attempt to egister these folders as sourceFOlders in idea.. but it has no effect.. it seems to work without it thanks to a different feature i found, at least in my testing on the windows version of idea